[Docs] Update return types of math functions

Blargian opened the pull request:
Changes return type of `tanh` from incorrect `Float64` to correct `Float32`

### Changelog category (leave one):
- Documentation (changelog entry is not required)

Blargian committed:
Correct `tanh` return type

Change return type from incorrect Float64 to correct Float32

rschu1ze commented:
@Blargian Thanks.

`SELECT toTypeName(tanh(toFloat64(0)));` returns a `Float64`.
`SELECT toTypeName(tanh(toFloat32(0)));` returns a `Float32`.

To be precise, we'd need to make a distinction based on the input type.

But it would be even better to look at the source code and find out how the return type is actually determined.

Side note: Ideally, we also double-check the documented return types of the [other math functions](https://clickhouse.com/docs/en/sql-reference/functions/math-functions) in ClickHouse.

Blargian commented:
> @Blargian Thanks.
> 
> `SELECT toTypeName(tanh(toFloat64(0)));` returns a `Float64`. `SELECT toTypeName(tanh(toFloat32(0)));` returns a `Float32`.
> 
> To be precise, we'd need to make a distinction based on the input type.
> 
> But it would be even better to look at the source code and find out how the return type is actually determined.
> 
> Side note: Ideally, we also double-check the documented return types of the [other math functions](https://clickhouse.com/docs/en/sql-reference/functions/math-functions) in ClickHouse.

@rschu1ze thanks for clarifying. I will go through and check all math functions.

Blargian committed:
Update math-functions argument types.

Many functions have argument type listed as `Float64` type when it can be `Float*`,`(U)Int*` or in some cases `Decimal*`

Blargian commented:
- [X] Update argument types
- [ ] Check code for how return is determined
